packages/starknet-snap/src/createAccount.ts

```diff
@@ -24,11 +28,15 @@ import { logger } from './utils/logger';
  * @param silentMode - The flag to disable the confirmation dialog from snap.
  * @param waitMode - The flag to enable an determination by doing an recursive fetch to check if the deploy account status is on L2 or not. The wait mode is only useful when it compose with other txn together, it can make sure the deploy txn execute complete, avoiding the latter txn failed.
  */
-export async function createAccount(params: ApiParams, silentMode = false, waitMode = false) {
```

Comment:
the End of this function is write a data in State for user, if you are setting require deploy & upgrade to true, that function may override